src/js/utils.js

```diff
@@ -322,6 +322,35 @@ function parseCookie(str, opts) {
   return parsed;
 }
 
+function CheckedDNTBuffer(size) {
+  if (size) {
+    this.size = size;
+  }
+  this.index = 0;
+  this.array = new Array(this.size);
+  this.set = new Set();
+}
+
+CheckedDNTBuffer.prototype = {
+  size: 1000,
+  // if we've seen item, return true. if we haven't store it and return false
+  hasOrSet: function(item) {
```

Comment:
I find the name `hasOrSet` to be a tad confusing. 

Could it instead be split into two functions? One being `has()`, which checks if a domain has already been checked. Another which is `add()`, which is called in the case that `_checkPrivacyBadgerPolicy()` returns `false` to the callback. 

I believe this would also eliminate the need for an explicit `delete()` function, as we wouldn't be adding anything that might need to be removed.